oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/CompositeIndexer.java

```diff
@@ -86,6 +87,13 @@ public Set<String> getRelativeIndexedNodeNames() {
         return result;
     }
 
+    @Override
+    public String getIndexName() {
+        // This is susceptible to infinite recursion if a composite indexer contains itself, directly or indirectly.
```

Comment:
```suggestion
        // Many methods are susceptible to infinite recursion if a composite indexer contains itself, directly or indirectly.
        // In this case, the methods would throw a StackOverflowException
```

Reply:
Yes, I agree, it should be a warning in the public documentation of the class. The users of this class have to be aware to avoid this issue. I updated the comment.